aes/src/armv8.rs

```diff
@@ -172,6 +175,16 @@ macro_rules! define_aes_impl {
             }
         }
 
+        impl Drop for $name_enc {
+            fn drop(&mut self) {
```

Comment:
It could be worth to add `#[inline]` attribute here and in other places. Otherwise, without LTO Rust will not be able to inline it.

Reply:
As far as I'm aware the only difference between implementing `Drop` and not implementing it is being able to implement `Copy`. The [reference](https://doc.rust-lang.org/stable/reference/special-types-and-traits.html?highlight=drop#drop) also does not seem to say much more about that.

The only other differences I could think of are performance issues, such as not having to track drop flags and not running an empty destructor (though inlining should prevent that). I added the inline attribute to all `drop` functions now.

In general I agree with @tarcieri that conditionally implementing `Drop` (or any trait) based on a not obviously related feature could lead to problems.

Reply:
If nothing else the bounds are different, although whether `Drop` bounds should be used/allowed at all is a matter of debate. Nevertheless, they do.

The other major difference is it's not possible to move out of a struct which impls `Drop`, although that's not happening in the current implementation so it's not a major issue.

I think `Drop` is magical enough conditionally dropping based on whether or not features are impl'd is a bit weird.